ansible/roles/influxdb_server/defaults/main.yml

```diff
@@ -0,0 +1,382 @@
+      - ca-certs: '"{{ influxdb_server__pki_path + "/" + influxdb_server__pki_realm +
+                       "/" + influxdb_server__pki_ca }}"'
+
+                                                                   # ]]]
+# .. envvar:: influxdb_server__pki [[[
+#
+# Enable or disable support for SSL in InfluxDB (using :ref:`debops.pki`).
+influxdb_server__pki: '{{ (True
+                          if (ansible_local|d() and ansible_local.pki|d() and
+                              ansible_local.pki.enabled|d() and
+                              influxdb_server__pki_realm in ansible_local.pki.known_realms)
+                          else False) | bool }}'
+
+                                                                   # ]]]
+# .. envvar:: influxdb_server__pki_path [[[
+#
+# Base path for PKI directory.
+influxdb_server__pki_path: '{{ (ansible_local.pki.base_path
+                               if (ansible_local|d() and ansible_local.pki|d() and
+                                   ansible_local.pki.base_path|d())
+                               else "/etc/pki") }}'
+
+                                                                   # ]]]
+# .. envvar:: influxdb_server__pki_realm [[[
+#
+# Default PKI realm used by InfluxDB server.
+influxdb_server__pki_realm: '{{ ansible_local.pki.realm
+                                if (ansible_local|d() and ansible_local.pki|d() and
+                                    ansible_local.pki.realm|d())
+                                else "domain" }}'
+
+                                                                   # ]]]
+# .. envvar:: influxdb_server__pki_ca [[[
+#
+# Root CA certificate used by InfluxDB, relative to :envvar:`influxdb_server__pki_realm`.
+influxdb_server__pki_ca: 'CA.crt'
+
+                                                                   # ]]]
+# .. envvar:: influxdb_server__pki_crt [[[
+#
+# Host certificate used by InfluxDB, relative to :envvar:`influxdb_server__pki_realm`.
+influxdb_server__pki_crt: 'default.crt'
+
+                                                                   # ]]]
+# .. envvar:: influxdb_server__pki_key [[[
+#
+# Host private key used by InfluxDB, relative to :envvar:`influxdb_server__pki_realm`.
+influxdb_server__pki_key: 'default.key'
+
+                                                                   # ]]]
+                                                                   # ]]]
+# Default root account [[[
+# ----------------------------
+
+# .. envvar:: influxdb_server__password_length [[[
+#
+# Length of automatically generated user accounts, saved in the ``secret/``
+# directory. See :ref:`debops.secret` role for more details about passwords.
+influxdb_server__password_length: '48'
+
+                                                                   # ]]]
+# .. envvar:: influxdb_server__root_password [[[
+#
+# Password for default root admin user.
+influxdb_server__root_password: '{{ lookup("password", secret + "/influxdb/" + ansible_fqdn +
```

Comment:
As I know the recommendation  where is to use ansible_hostname. Right @drybjed ?

Reply:
No, that would be `inventory_hostname` instead.

Reply:
It's a problem.

If I use `inventory_hostname`, it generate the password in `secret/influxdb/influxdb-ams3-01/credentials/root/password` path.

But `influxdb` role use this configuration for search root password:

```password: '{{ lookup("password",
                  secret + "/influxdb/" + influxdb__delegate_to +
                  "/credentials/root/password " +
                  "length=" + influxdb__password_length) }}'

# .. envvar:: influxdb__server [[[
#
# FQDN hostname of the InfluxDB server. If a local InfluxDB server is detected, it
# will override this variable automatically. Only one server at a time is
# currently supported per host.
influxdb__server: ''

# .. envvar:: influxdb__delegate_to [[[
#
# When the InfluxDB server is used remotely, Ansible needs to run tasks on the
# correct host. This variable controls the task delegation to the correct
# database server.
#
# If the InfluxDB server is run locally, this should point to the inventory name
# of the current host, NOT ``localhost`` because that would delegate the tasks
# to the Ansible Controller.
influxdb__delegate_to: '{{ influxdb__server
                           if (influxdb__server|d() and
                               influxdb__server != "localhost")
                           else ansible_fqdn }}'
```
For remote connections `influxdb__server` is a FQDN, like `influxdb__server: 'influxdb-ams3-01.neuromobile.io'`, other roles can use `influxdb__server` var for configure connections to InfluxDB database. And `influxdb__delegate_to` will be configured with `influxdb__server` value.

So, it will search the root password in a incorrect path: `secret/influxdb/influxdb-ams3-01.neuromobile.io/credentials/root/password`.

Reply:
In other database roles use local accounts, but influxdb use a HTTP API to connect and needs found root password in the correct path.

Reply:
I see, so the DNS name is required in that case. That explains things. :-) you should mention that in the documentation just in case.

In this special case (REST API) I'm not sure if delegating the task to the InfluxDB server host has any sense, since the task will be ran over HTTP, right? So `delegate_to` functionality can probably be removed.

Reply:
I would keep `delegate_to`. It can be useful when you have some firewalls between Ansible controller and the host. I have some environments where I use `ssh` with several jumphosts to access servers managed with DebOps and I'm not able to access InfluxDB port directly.

Reply:
Upgraded documentation